packages/db-mongodb/src/queries/buildSortParam.ts

```diff
@@ -18,12 +32,84 @@ export type SortArgs = {
 
 export type SortDirection = 'asc' | 'desc'
 
+const relationshipSort = ({
+  adapter,
+  fields,
+  path,
+  sort,
+  sortAggregation,
+  sortDirection,
+}: {
+  adapter: MongooseAdapter
+  fields: FlattenedField[]
+  path: string
+  sort: Record<string, string>
+  sortAggregation: PipelineStage[]
+  sortDirection: SortDirection
+}) => {
+  let currentFields = fields
+  const segments = path.split('.')
+  if (segments.length < 2) {
+    return false
+  }
+
+  for (const [i, segment] of segments.entries()) {
+    const field = currentFields.find((each) => each.name === segment)
+
+    if (!field) {
+      return false
+    }
+
+    if ('fields' in field) {
+      currentFields = field.flattenedFields
+    } else if (field.type === 'relationship' && i !== segments.length - 1) {
```

Comment:
I think we can change this to support upload field type too. 
`field.type === 'relationship' || field.type === 'upload'`